nucypher/blockchain/eth/signers.py

```diff
@@ -231,3 +253,100 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+    """Local Web3 signer implementation supporting a single account/keystore file"""
+
+    IS_SIGNER_LOCAL = True
+
+    __key: Dict
```

Comment:
Woah - Is this the dataclass syntax :-)  

This means we'd be committing to not supporting 3.6 (only 3.7+), correct?

<https://www.python.org/dev/peps/pep-0557/>

Reply:
Uhm... I'd still be supporting 3.6, tbh

Reply:
Those are instance variable type hints in the class body as per PEP 526, no \@dataclass decorator.

https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations

For what it's worth I use the stock python on Centos 7 which is 3.6.8; I haven't noticed any issues with it:
```
Python 3.6.8 (default, Apr 25 2019, 21:02:35)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
```

Reply:
Ah, yeah I see - They look similar!